File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys, math
input = sys.stdin.readline
sys.setrecursionlimit(10 ** 6)

S = input().rstrip()
B = input().rstrip()

S = S[::-1]
stack = []

i = 0
while i < len(S):
    stack.append(S[i])
    if "".join(stack[-1:-1-len(B):-1]) == B:
        for k in range(0, len(B)):
            stack.pop(-1)
    i += 1

if stack:
    print("".join(stack[::-1]))
else:
    print("FRULA")

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')
// let input = fs.readFileSync('./EX.txt').toString().trim().split('\n')

const S = input[0].trim()
const B = input[1].trim()


let stack = []
let start = 0
for (let i = 0; i < S.length; i++) {
    stack.push(S[i])
    
    if (stack.slice(stack.length-B.length,stack.length).join("") === B) {
        stack.splice(stack.length-B.length, B.length)
    }

}
if (stack.length) {
    console.log(stack.join(""))
} else {
    console.log("FRULA")
}
